### PR TITLE
Newsletters: add 80 (2019-01-13)

### PR DESCRIPTION
**Pull request by harding**

### CODE REVIEW

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    As an alternative to either BIP9 or BIP8 alone, Corallo proposes
+    a three-step process:  use BIP9 to allow a proposal to be
+    activated within a one-year window; pause for a six month discussion
+    period if the proposal is not activated; and---if it's clear that
+    the community still wants the proposal activated---force activation
+    using a BIP8 flag-day set to two years in the future (with faster
+    activation possible using versionbits signaling).  Node software can
+    prepare for this maximum 42 month process by including, even in its
+    initial versions, a configuration option users can manually enable
+    to enfoce the BIP8 flag day if necessary.  If the first 18 months of
```
Comment (bitschmidty) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: s/enfoce/enforce

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [Bitcoin Core #17578][] updates the `getaddressinfo` RPC to return an
+  array of strings corresponding to the labels used by an address.
+  Previously, it returned an array of objects where each label contained
+  fields for its name and its purpose, where purpose corresponded to the
+  part of the wallet that generated the address.  Users who need the old
+  behavior can used the `-deprecatedrpc=labelspurpose` configuration
```
Comment (bitschmidty) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: s/can used/can use

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+- [Bitcoin Core #16373][] causes the `bumpfee` RPC used for
+  [Replace-by-Fee][topic rbf] fee bumping to return a [Partially Signed
+  Bitcoin Transaction][topic psbt] (PSBT) when the user attempts to fee
+  bump a transaction using a wallet with disabled private keys.  The
+  PSBT can then be copied to an external wallet (such as a hardware
+  device or cold wallet) for signing.
+
+- [Bitcoin Core #17621][] fixes a potential privacy leak for users of
+  the the `avoid_reuse` wallet flag.  This flag prevents the wallet from
```
Comment (bitschmidty) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: s/the the/the

**Comment by bitschmidty** [comment]
Ive looked into the failing of these checks and was able to force another netlify build which seemed to work. I suspect the next push to this branch will resolve the ❌ s

**Comment by harding** [comment]
Made edits for @bitschmidty feedback (thanks!).  Force pushed because I had to correct a typo in the commit message anyway.

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    mandatory versionbits signaling leading up to the flag day (similar
+    to how [BIP148][] proposed to activate segwit); Corallo
+    [notes][corallo reply timon] that this conflicts with the third and
+    fourth goals.  Jeremy Rubin provides a [quick analysis][rubin sf] of
+    his previous [spork proposal][spork vid] (see [Newsletter #32][spork
+    summary]) in the context of the five goals.  Anthony Towns provides
+    [lucid commentary][towns sf] on several aspects of Corallo's
+    proposal and Timón's response.
+
+    No clear conclusion was reached in the thread and we suspect
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: style nit: I think *expect* is better than *suspect* here

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  [Replace-by-Fee][topic rbf] fee bumping to return a [Partially Signed
+  Bitcoin Transaction][topic psbt] (PSBT) when the user attempts to fee
+  bump a transaction using a wallet with disabled private keys.  The
+  PSBT can then be copied to an external wallet (such as a hardware
+  device or cold wallet) for signing.
+
+- [Bitcoin Core #17621][] fixes a potential privacy leak for users of
+  the `avoid_reuse` wallet flag.  This flag prevents the wallet from
+  spending bitcoins received to an address the wallet has already used
+  for spending (see [Newsletter #52][]).  This enhances privacy by
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: Link directly to en/newsletters/2019/06/26/#bitcoin-core-13756 ?

Comment (jnewbery) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: style nit: Starting two consecutive sentences with "This ..." seems slightly inelegant to me.

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+  preventing otherwise-unrelated transactions from being associated
+  together because they all involve the same address.  However, Bitcoin
+  Core currently monitors for payments to any of its public keys using
+  any of several different address formats.  This meant that multiple
+  payments to the same public key---but with different addresses (e.g.
+  P2PKH, P2WKH, P2SH-P2WPKH)---could be associated with each other on
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: s/P2WKH/P2WPKH/

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+  the block chain even though the `avoid_reuse` behavior thought it
```
Comment (jnewbery) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: "the avoid_reuse behavior thought..." seems a bit imprecise to me. It's the user who thinks he/she isn't reusing addresses. The "avoid_reuse behaviour" can't have thoughts!

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+  for spending (see [Newsletter #52][news52 avoid_reuse]).  This enhances privacy by
+  preventing otherwise-unrelated transactions from being associated
+  together because they all involve the same address.  However, Bitcoin
+  Core currently monitors for payments to any of its public keys using
+  any of several different address formats.  This meant that multiple
+  payments to the same public key---but with different addresses (e.g.
+  P2PKH, P2WPKH, P2SH-P2WPKH)---could be associated with each other on
+  the block chain even though the `avoid_reuse` behavior is supposed to
+  prevent this type of linkability.  This merged PR fixes the problem for the
+  `avoid_reuse` flag and the ongoing adoption of [output script
```
Comment (jonatack) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: nit: s/flag/flag,/ (or make two sentences) for clarity

Comment (jonatack) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: perhaps worth mentioning that this PR is being backported in https://github.com/bitcoin/bitcoin/pull/17792 for release in 0.19.1

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+  of the next major version of LND brings several new features and bug
+  fixes.  Experienced users are encouraged to help test the software so
+  that any problems can be identified and fixed prior to release.
+
+## News
+
+- **Discussion of soft fork activation mechanisms:** Matt Corallo
+  started a [discussion][corallo sf] on the Bitcoin-Dev mailing list about
+  what attributes are desirable in a soft fork activation method and
+  submitted a proposal for a mechanism that contains those attributes.
```
Comment (jonatack) [inline _posts/en/newsletters/2020-01-15-newsletter.md]: FWIW I followed this ML exchange with interest, but wondered what the context was for launching the discussion. Was there an event, a trigger, something in particular?

Comment (harding) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: There was a bit of discussion about what upgrade mechanism to use for taproot a few weeks ago; this didn't involve Matt (although he was mentioned): http://www.erisian.com.au/meetbot/taproot-bip-review/2019/taproot-bip-review.2019-12-17-19.01.log.html#l-56

But I don't know for sure whether that had anything to do with Matt's post, so I hesitate to link to it.

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 1 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+  eliminate the problem of Bitcoin Core monitoring for alternative
+  addresses.
+
+- [LND #3829][] makes a number of internal changes and documentation
+  updates in order to simplify adding [anchor outputs][] in the future.
+
+{% include references.md %}
+{% include linkers/issues.md issues="17578,16373,17621,3829" %}
+[lnd 0.9.0-beta]: https://github.com/lightningnetwork/lnd/releases/tag/v0.9.0-beta-rc1
+[corallo sf]: https://github.com/lightningnetwork/lnd/releases/tag/v0.9.0-beta-rc1
```
[the remark this answers is not in the document]
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 2 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 3 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 4 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: changed

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 5 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: done

Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 6 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 7 above)
Comment (jnewbery) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: fixed

**Comment by jnewbery** [comment]
ACK 182cbabe873c1b2fec94c7124dbdbfe529719042. I've marked all resolved comments as resolved

File: _posts/en/newsletters/2020-01-15-newsletter.md (hunk 9 above)
Comment (jonatack) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: Thanks @harding. I agree that it's too tenuous to link to it.

File: _posts/en/newsletters/2020-01-15-newsletter.md [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+    As an alternative to either BIP9 or BIP8 alone, Corallo proposes
+    a three-step process:  use BIP9 to allow a proposal to be
+    activated within a one-year window; pause for a six month discussion
+    period if the proposal is not activated; and---if it's clear that
+    the community still wants the proposal activated---force activation
+    using a BIP8 flag-day set to two years in the future (with faster
+    activation possible using versionbits signaling).  Node software can
+    prepare for this maximum 42 month process by including, even in its
+    initial versions, a configuration option users can manually enable
+    to enforce the BIP8 flag day if necessary.  If the first 18 months of
```
[the remark this answers is not in the document]
Comment (harding) [reply _posts/en/newsletters/2020-01-15-newsletter.md]: Good catch.  One of the uses is "flag-day soft fork".  I'm leaving the hyphen there and removing it in the other non-modifier usecases.  Thanks!

**Comment by jonatack** [comment]
ACK be6749e1b0b8794b04dc372ae173d33a55f7a0da edits

**Comment by adamjonas** [comment]
ACK be6749e1b0b8794b04dc372ae173d33a55f7a0da

**Comment by bitschmidty** [comment]
ACK be6749e1b0b8794b04dc372ae173d33a55f7a0da

Squashed and merged

100th bitdevs happened in New York
Matt Corallo talking the soft fork

Changes in #16373
To bumpfee, RPC
For Replace-by-Fee
Returns a PSBT
When disabled private key
Use a floppy
To your wallet copy

**Pull request merged**